cmd/service/dev.go

```diff
@@ -28,7 +29,10 @@ func init() {
 }
 
 func devHandler(cmd *cobra.Command, args []string) {
-	serviceID, err := createService(defaultPath(args))
+	serviceID, isValid, err := createService(defaultPath(args))
+	if !isValid {
+		os.Exit(1)
```

Comment:
In the gRPC side, validation errors send as a part of gRPC [response](https://github.com/mesg-foundation/core/blob/b17faa6/api/core/api.proto#L238) stream. But server-side errors of core returned/send as an error in gRPC handlers as we do always. In this case I like separating both of them like this. Validation errors should not be returned as a gRPC handler error because they are different kind of errors that needed to be take action on in the client side. So to identify them as a separate kind we need to split them like the current implementation.

_isValid_ set to _false_ when there is a validation error sent by server [here](https://github.com/mesg-foundation/core/blob/5e7fd59/cmd/service/deploy.go#L140) and as you can see there is some _Print_ function above that line to let devs know there is a validation error. If there is a validation error, we should exit with 1 after we printed those messages but i quite don't like exiting in goroutine because this makes things un-trackable. So instead we send isValid false value through a channel which is read within _ deployService()_ and returned as its return value. so callers of deployService() should check for isValid and immediately exit with 1. If you don't like this approach we can change isValid as _validationErrors_ and fill it with validation errors instead. And callers of deployService() can check it and print its errors with a util function like 	_utils.HandleValidationError(validationError)_. So this function can print the validation error and exit with 1.

For the second question, _err_ return value of deployService() still filled with server errors and we check them in the code with _utils.HandlerError(err)_. I don't see a problem here.